src/flask_smorest/exceptions.py

```diff
@@ -7,30 +7,166 @@ class FlaskSmorestError(Exception):
+    """Conflict"""
+
+
+class Gone(wexc.Gone, ApiException):
+    """Resource gone"""
+
+
+class LengthRequired(wexc.LengthRequired, ApiException):
+    """Length required"""
+
+
+class PreconditionFailed(wexc.PreconditionFailed, ApiException):
+    """Etag required and wrong ETag provided"""
+
+
+class RequestEntityTooLarge(wexc.RequestEntityTooLarge, ApiException):
+    """Request entity too large"""
+
+
+class RequestURITooLarge(wexc.RequestURITooLarge, ApiException):
+    """Request URI too large"""
+
+
+class UnsupportedMediaType(wexc.UnsupportedMediaType, ApiException):
+    """Unsupported media type"""
+
+
+class RequestedRangeNotSatisfiable(wexc.RequestedRangeNotSatisfiable, ApiException):
+    """Requested range not satisfiable"""
 
 
-class PreconditionRequired(wexc.PreconditionRequired, FlaskSmorestError):
+class ExpectationFailed(wexc.ExpectationFailed, ApiException):
+    """Expectation failed"""
+
+
+class ImATeapot(wexc.ImATeapot, ApiException):
+    """I'm a teapot"""
+
+
+class UnprocessableEntity(wexc.UnprocessableEntity, ApiException):
+    """Unprocessable entity"""
+
+
+class Locked(wexc.Locked, ApiException):
+    """Locked"""
+
+
+class FailedDependency(wexc.FailedDependency, ApiException):
+    """Failed dependency"""
+
+
+class PreconditionRequired(wexc.PreconditionRequired, ApiException):
     """Etag required but missing"""
 
     # Overriding description as we don't provide If-Unmodified-Since
     description = 'This request is required to be conditional; try using "If-Match".'
 
 
-class PreconditionFailed(wexc.PreconditionFailed, FlaskSmorestError):
-    """Etag required and wrong ETag provided"""
+class TooManyRequests(wexc.TooManyRequests, ApiException):
+    """Too many requests"""
 
 
-class CurrentApiNotAvailableError(FlaskSmorestError):
-    """`current_api` not available"""
+class RequestHeaderFieldsTooLarge(wexc.RequestHeaderFieldsTooLarge, ApiException):
+    """Request header fields too large"""
+
+
+class UnavailableForLegalReasons(wexc.UnavailableForLegalReasons, ApiException):
+    """Unavailable for legal reasons"""
+
+
+class InternalServerError(wexc.InternalServerError, ApiException):
+    """Internal server error"""
+
+
+class NotImplemented(wexc.NotImplemented, ApiException):
+    """Not implemented"""
+
+
+class BadGateway(wexc.BadGateway, ApiException):
+    """Bad gateway"""
+
+
+class ServiceUnavailable(wexc.ServiceUnavailable, ApiException):
+    """Service unavailable"""
+
+
+class GatewayTimeout(wexc.GatewayTimeout, ApiException):
+    """Gateway timeout"""
+
+
+class HTTPVersionNotSupported(wexc.HTTPVersionNotSupported, ApiException):
+    """HTTP version not supported"""
+
+
+class NotModified(ApiException):
+    """Resource was not modified (Etag is unchanged)
+
+    Exception created to compensate for a lack in Werkzeug (and Flask)
+    """
+
+    code = 304
+    description = "Resource not modified since last request."
+
+
+def abort(http_status_code, exc=None, **kwargs):
```

Comment:
I like to keep only exceptions in this file.

We may want to put this somewhere else. It could be in `error_handler.py`.

Reply:
I'll fix that import path, I think it's a good idea to fix it as you said